client/src/app/site/assignments/components/assignment-detail/assignment-detail.component.ts

```diff
@@ -210,12 +217,36 @@ export class AssignmentDetailComponent extends BaseViewComponentDirective implem
      * Init data
      */
     public ngOnInit(): void {
+        this.observeRoute();
         this.getAssignmentByUrl();
         this.agendaObserver = this.itemRepo.getViewModelListBehaviorSubject();
         this.tagsObserver = this.tagRepo.getViewModelListBehaviorSubject();
         this.mediafilesObserver = this.mediafileRepo.getViewModelListBehaviorSubject();
     }
 
+    /**
+     * Called during view destruction.
+     */
+    public ngOnDestroy(): void {
+        if (this.navigationSubscription) {
+            this.navigationSubscription.unsubscribe();
+        }
+        super.ngOnDestroy();
+    }
+
+    /**
+     * Observes the route for events. Calls to clean all subs if the route changes.
+     * Calls the assignment details from the new route.
+     */
+    public observeRoute(): void {
```

Comment:
should be private

Reply:
Makes sense. I blindly copied this from 
[motion-detail](https://github.com/OpenSlides/OpenSlides/blob/master/client/src/app/site/motions/modules/motion-detail/components/motion-detail/motion-detail.component.ts#L609)

Reply:
That's what I thought. Should also be private there.

Reply:
Commit has been added.